pybaseball/league_pitching_stats.py

```diff
@@ -42,15 +43,16 @@ def get_soup(start_dt, end_dt):
     if((start_dt is None) or (end_dt is None)):
         print('Error: a date range needs to be specified')
         return None
-    url = "http://www.baseball-reference.com/leagues/daily.cgi?user_team=&bust_cache=&type=p&lastndays=7&dates=fromandto&fromandto={}.{}&level=mlb&franch=&stat=&stat_value=0".format(start_dt, end_dt)
+    url = f"https://www.fangraphs.com/leaders.aspx?pos=all&stats=pit&lg=all&qual=1&type=c,4,5,6,7,8,9,10,11,114,12,13,14,15,16,17,18,19,20,21,22,23,24,36,37,38,40,120,121,217,41,42,43,44,117,118,119,45,124,62,122,229,240,251,262,273,230,241,252,263,274,231,242,253,264,275,234,245,256,267,278,226,237,248,259,270,235,246,257,268,279,228,239,250,261,272,227,238,249,260,271,232,243,254,265,276,233,244,255,266,277,236,247,258,269,280,63,64,65,66,67,68,69,70,71,72,73,74,115,116,212,213,214,215,58,59,60,54,55,56,57,29,30,31,292,293,294,295,296,297,298,299&season={start_dt[:4]}&month=0&season1={end_dt[:4]}&ind=0&team=0&rost=0&age=0&filter=&players=0&startdate={start_dt}&enddate={end_dt}&page=1_1500"
     s = requests.get(url).content
     return BeautifulSoup(s, "lxml")
 
 
 def get_table(soup):
-    table = soup.find_all('table')[0]
+    warnings.warn("\nPlease consider supporting FanGraphs\nhttps://plus.fangraphs.com/product/fangraphs-membership/", Warning)
```

Comment:
Okay I'm not _sure_ about this one

Reply:
Why? They offer a free service and ask for membership. Seems fair if we're scraping to at least promote membership in good faith

Reply:
Yeah, but if so, I would suggest doing this with print, not warnings, which has implications here that are incorrect (that something went wrong or is otherwise liable to break).
And if we're going to have a message, we might as well put in 'Downloading from FanGraphs' before that.

Reply:
Well that's the problem isn't it? If this goes behind a pay wall like stuff on bbref, it's going to break? Unless you're getting a tap from MLBAM (and with the current gigantic warnings on FG saying they need support to maintain the site), we can't really trust this source will stay the same. That's not different from bbref scraping, except they've figured out a way to monetize at scale so maybe things stay public (or maybe they start tucking more things behind a pay wall). But with FG it seems as though they want to keep things as open as possible, and this is a painless way to continue keeping their service open. 

Technically speaking, I suppose I don't care about warning vs print but a warning seems more appropriate for the above reasons.